### PR TITLE
Enforce current linting rules

### PR DESCRIPTION
**Pull request by dpyro**
Currently this project does not pass a `lint` check. I cleaned up the code following the current lint rules.

### CODE REVIEW

**Comment by rynomster** [comment]
I think if lint passes it'd also be a good idea to uncomment it in the Travis build, and then change the repository settings to only allow PR merges if all checks pass. I think those checks help allot with review time.

File: .eslintrc [hunk 1]
```diff
@@ -1,11 +1,10 @@
 {
   "env": {
```
Comment (spresnal) [inline .eslintrc]: Would be nice to enforce ES6

File: lib/transactions/crypto.js [hunk 2]
```diff
@@ -71,22 +69,22 @@ function getBytes(transaction, skipSignature, skipSecondSignature) {
...
 	}
 
-	var bb = new ByteBuffer(1 + 4 + 32 + 8 + 8 + 21 + 64 + 64 + 64 + assetSize, true);
+	bb = new ByteBuffer(1 + 4 + 32 + 8 + 8 + 21 + 64 + 64 + 64 + assetSize, true);
 	bb.writeByte(transaction.type);
 	bb.writeInt(transaction.timestamp);
 
 	var senderPublicKeyBuffer = new Buffer(transaction.senderPublicKey, "hex");
-	for (var i = 0; i < senderPublicKeyBuffer.length; i++) {
+	for (i = 0; i < senderPublicKeyBuffer.length; i++) {
```
Comment (spresnal) [inline lib/transactions/crypto.js]: My preference would lean towards using let in each of the for loops versus defining it at the top using var

File: lib/transactions/crypto.js [hunk 3]
```diff
@@ -212,7 +210,7 @@ function fromBytes(hexString){
 		delete tx.recipientId;
 		parseSignatures(hexString, tx, 76+42+128+32);
 	}
-	console.log(tx);
+	//console.log(tx);
```
Comment (spresnal) [inline lib/transactions/crypto.js]: Probably just delete this

File: test/ecpair/index.js [hunk 4]
```diff
@@ -234,12 +231,12 @@ describe('ECPair', function () {
     })
 
     describe('verify', function () {
```
Comment (spresnal) [inline test/ecpair/index.js]: Might as well comment out the entire test vs having the shell run as if it's doing something

File: .eslintrc (hunk 1 above)
Comment (spresnal) [reply .eslintrc]: If we do go that route it'd be preferred to use const for the top level variables & let for scoped blocks (might be a change for another PR)

Comment (dpyro) [reply .eslintrc]: ES6 will require transpiling support or bumping up the minimum JS implementation.

File: lib/transactions/crypto.js (hunk 2 above)
Comment (dpyro) [reply lib/transactions/crypto.js]: Would need to go into a different PR.

File: lib/transactions/crypto.js [hunk 5]
```diff
@@ -71,22 +69,22 @@ function getBytes(transaction, skipSignature, skipSecondSignature) {
 
 	}
 
-	var bb = new ByteBuffer(1 + 4 + 32 + 8 + 8 + 21 + 64 + 64 + 64 + assetSize, true);
+	bb = new ByteBuffer(1 + 4 + 32 + 8 + 8 + 21 + 64 + 64 + 64 + assetSize, true);
```
Comment (luciorubeens) [inline lib/transactions/crypto.js]: What if the condition does not reach 4?

File: lib/transactions/crypto.js [hunk 6]
```diff
@@ -102,8 +100,8 @@ function getBytes(transaction, skipSignature, skipSecondSignature) {
 		}
 	}
 	else if (transaction.vendorField) {
-		var vf = new Buffer(transaction.vendorField);
-		var fillstart=vf.length;
+		vf = new Buffer(transaction.vendorField);
```
Comment (luciorubeens) [inline lib/transactions/crypto.js]: these variables need not be defined before the scope?

```
let vf;
if (true) vf = true;
else vf = false;
```

Comment (dpyro) [reply lib/transactions/crypto.js]: `var` isn't blocked-scoped in JavaScript so I don't think they need to be defined ahead in the file. I moved the `var` defines ahead anyway to improve readability.

File: lib/transactions/crypto.js (hunk 5 above)
Comment (dpyro) [reply lib/transactions/crypto.js]: It may still work because `var` is weird but because I'm not sure I moved it out anyway. 😅

**Comment by faustbrian** [comment]
Will close this as we started working on v2.

**Pull request closed without merging**